t/lib/t/MusicBrainz/Server/Data/Area.pm

```diff
@@ -58,33 +58,35 @@ EOSQL
     is($area->name, 'descendant', 'correct descendant country is loaded');
 
     $test->c->model('Area')->load_containment($area);
-    is($area->parent_city->name, 'parent city', 'correct parent city is loaded');
-    is($area->parent_subdivision->name, 'parent subdivision', 'correct parent subdivision is loaded');
-    is($area->parent_country->name, 'parent country', 'correct parent country is loaded');
+    is(scalar @{$area->containment}, 4);
+    is($area->containment->[0]->name, 'parent city', 'correct parent city is loaded');
+    is($area->containment->[1]->name, 'parent subdivision', 'correct parent subdivision is loaded');
+    is($area->containment->[2]->name, 'parent country', 'correct parent country is loaded');
+    is($area->containment->[3]->name, 'parent meta-country', 'parent meta-country is loaded');
 
     $area = $test->c->model('Area')->get_by_id(2);
     is($area->name, 'parent city', 'correct descendant is loaded');
 
     $test->c->model('Area')->load_containment($area);
-    is($area->parent_city, undef, 'no parent city is loaded');
-    is($area->parent_subdivision->name, 'parent subdivision', 'correct parent subdivision is loaded');
-    is($area->parent_country->name, 'parent country', 'correct parent country is loaded');
+    is(scalar @{$area->containment}, 3);
+    is($area->containment->[0]->name, 'parent subdivision', 'correct parent subdivision is loaded');
+    is($area->containment->[1]->name, 'parent country', 'correct parent country is loaded');
+    is($area->containment->[2]->name, 'parent meta-country', 'parent meta-country is loaded');
```

Comment:
This is the one change in behavior I can see: the containment array holds the entire hierarchy of areas with type in (1, 2, 3), so if there's a parent and parent-parent country, they'll both get included in the containment.